## BEMM458 Final Assignment

## You are a data scientist working for the European Union. There is an increasing concern inn  relation to the spread and amplification of misinformation on social media.

## General management needs you to conduct an analysis of recent conversations posted on Twitter in relation to spain. You have to identify ongoing misinformation campaigns and profile those users contributing to the spread of misinformation.




In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import scipy as scipy
import re

In [2]:
from IPython.core.display import display, HTML
display(HTML("<style>.container { width:100% !important; }</style>"))

In [3]:
pd.set_option('display.max_rows', 120)

In [4]:
pd.set_option('display.float_format', '{:.0f}'.format)

In [5]:
spainEnglishAll=pd.read_csv('./BEMM458FinalAssignment.csv')

In [6]:
spainEnglishAll.columns

Index(['Unnamed: 0', 'tweeet.id', 'tweet.text', 'tweet.created_at',
       'tweet.author_id', 'tweet.lang', 'public_metrics.retweet_count',
       'public_metrics.reply_count', 'public_metrics.like_count'],
      dtype='object')

In [7]:
spainEnglishAll

Unnamed: 0            tweeet.id  \
0               0  1461484086892146692   
1               1  1461484079245840390   
2               2  1461484043187462149   
3               3  1461483982105763842   
4               4  1461483978658127872   
...           ...                  ...   
47545       47545  1461484434209783814   
47546       47546  1461484401712414722   
47547       47547  1461484391255977994   
47548       47548  1461484374558490625   
47549       47549  1461484366325035009   

                                              tweet.text  \
0      RT @ESPNFC: EURO 2008 top scorer: Villa ⚽️\nWo...   
1      RT @MrGeepGeep: SOE is over and now is the tim...   
2      RT @AnittaCrave: "Envolver" — by Anitta will d...   
3      Men jewelry  - Madrid, Spain \n@GuilleRBRB htt...   
4      RT @Foreign_Cat: #Turkey &amp; #Spain, partner...   
...                                                  ...   
47545  @only1womble @SimonJonesNews @pritipatel Someo...   
47546  RT @ancientorigins: A team of #scientists has ...   
47547  On #Friday Amazing time in #SabatiniGardens #M...   
47548  RT @THEMAXHILTON: The wait is over !  you have...   
47549  #Barcelona Nov 19 01:00 Temperature 10C Wind N...   

                tweet.created_at     tweet.author_id tweet.lang  \
0      2021-11-18 23:58:54+00:00          3239744421         en   
1      2021-11-18 23:58:52+00:00          2312771653         en   
2      2021-11-18 23:58:43+00:00 1296857652677550080         en   
3      2021-11-18 23:58:29+00:00 1071703322166480768         en   
4      2021-11-18 23:58:28+00:00 1142741725225652224         en   
...                          ...                 ...        ...   
47545  2021-11-19 00:00:16+00:00          1733373504         en   
47546  2021-11-19 00:00:09+00:00           100887413         en   
47547  2021-11-19 00:00:06+00:00  995441471938580352         en   
47548  2021-11-19 00:00:02+00:00  832015137263452160         en   
47549  2021-11-19 00:00:00+00:00           258025521         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
0                              1568                           0   
1                                38                           0   
2                                90                           0   
3                                 0                           0   
4                                34                           0   
...                             ...                         ...   
47545                             0                           0   
47546                            18                           0   
47547                             0                           0   
47548                           121                           0   
47549                             0                           0   

       public_metrics.like_count  
0                              0  
1                              0  
2                              0  
3                              5  
4                              0  
...                          ...  
47545                          0  
47546                          0  
47547                          1  
47548                          0  
47549                          0  

[47550 rows x 9 columns]

## Question 1. Which twitter conversations have large levels of amplification ? (15 points)
### Tip: consider number of retweets, likes, etc as proxies to measure the spread of information on social media

### <font color='green'> Answer (First Part): First, we need to find approximate  amplification relation for the retweets, reply and like counts. If our goal is the finding amplification of misinformation, we need to look which terms reach the tweet to the people's screen most. Retweet of tweet directly shows the tweet for all the followers of that account. Reply number should be less attribute to spreading misinformation, because the person who replied can also state that tweet has misinformation. However, reply count still very important at spreading of misinformation because replying moves that tweet to the person's followers' main screen. For like count, we can see some liked tweets of accounts that we are following in our twitter main screen, but frequency of it so low when it compared with other 2 parameters. However, like count has a meaning of confirmation on twitter and it can affect people's acceptance for that misinformation, so it has still some effect for amplification of misinformation. To conclude, according to their effects, we need assign coefficients to these terms and multiply with it. Then, we can sum up these counts and create new column as amplification_volume. There is no concrete coefficients for these terms, but even if we give bigger coefficient number according to their power of amplification, it would be more meaningful analyst rather than just count numbers.
Coefficient of Retweet : 5 --- Coefficient of Reply : 4 --- Coefficient of Like = 2 </font>

In [8]:
spainEnglishAll["amplification_volume"] = (spainEnglishAll["public_metrics.retweet_count"]*5) + (spainEnglishAll["public_metrics.like_count"] * 2) + (spainEnglishAll["public_metrics.reply_count"]*4)

In [9]:
spainEnglishAll

Unnamed: 0            tweeet.id  \
0               0  1461484086892146692   
1               1  1461484079245840390   
2               2  1461484043187462149   
3               3  1461483982105763842   
4               4  1461483978658127872   
...           ...                  ...   
47545       47545  1461484434209783814   
47546       47546  1461484401712414722   
47547       47547  1461484391255977994   
47548       47548  1461484374558490625   
47549       47549  1461484366325035009   

                                              tweet.text  \
0      RT @ESPNFC: EURO 2008 top scorer: Villa ⚽️\nWo...   
1      RT @MrGeepGeep: SOE is over and now is the tim...   
2      RT @AnittaCrave: "Envolver" — by Anitta will d...   
3      Men jewelry  - Madrid, Spain \n@GuilleRBRB htt...   
4      RT @Foreign_Cat: #Turkey &amp; #Spain, partner...   
...                                                  ...   
47545  @only1womble @SimonJonesNews @pritipatel Someo...   
47546  RT @ancientorigins: A team of #scientists has ...   
47547  On #Friday Amazing time in #SabatiniGardens #M...   
47548  RT @THEMAXHILTON: The wait is over !  you have...   
47549  #Barcelona Nov 19 01:00 Temperature 10C Wind N...   

                tweet.created_at     tweet.author_id tweet.lang  \
0      2021-11-18 23:58:54+00:00          3239744421         en   
1      2021-11-18 23:58:52+00:00          2312771653         en   
2      2021-11-18 23:58:43+00:00 1296857652677550080         en   
3      2021-11-18 23:58:29+00:00 1071703322166480768         en   
4      2021-11-18 23:58:28+00:00 1142741725225652224         en   
...                          ...                 ...        ...   
47545  2021-11-19 00:00:16+00:00          1733373504         en   
47546  2021-11-19 00:00:09+00:00           100887413         en   
47547  2021-11-19 00:00:06+00:00  995441471938580352         en   
47548  2021-11-19 00:00:02+00:00  832015137263452160         en   
47549  2021-11-19 00:00:00+00:00           258025521         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
0                              1568                           0   
1                                38                           0   
2                                90                           0   
3                                 0                           0   
4                                34                           0   
...                             ...                         ...   
47545                             0                           0   
47546                            18                           0   
47547                             0                           0   
47548                           121                           0   
47549                             0                           0   

       public_metrics.like_count  amplification_volume  
0                              0                  7840  
1                              0                   190  
2                              0                   450  
3                              5                    10  
4                              0                   170  
...                          ...                   ...  
47545                          0                     0  
47546                          0                    90  
47547                          1                     2  
47548                          0                   605  
47549                          0                     0  

[47550 rows x 10 columns]

In [10]:
amp_sorted = spainEnglishAll.sort_values("amplification_volume", ascending=False)

In [11]:
amp_sorted.head(100)

Unnamed: 0            tweeet.id  \
23467       23467  1461162679809155076   
23656       23656  1461157780664901633   
8820         8820  1461371211343020033   
39357       39357  1461649200882405377   
45373       45373  1461537918900600843   
41908       41908  1461613145219239944   
23655       23655  1461157789397438464   
45333       45333  1461539133520695297   
45374       45374  1461537902106537991   
44456       44456  1461565440879206401   
23465       23465  1461162702307426312   
45332       45332  1461539163207995397   
45336       45336  1461539044974731265   
45375       45375  1461537878991781892   
23658       23658  1461157760607739909   
12779       12779  1461327131057680386   
9806         9806  1461359523298377745   
23311       23311  1461167495809617921   
45334       45334  1461539103741149186   
23461       23461  1461162923057827840   
45378       45378  1461537845516984321   
41830       41830  1461614169296306178   
45337       45337  1461539030298886147   
45335       45335  1461539081725251585   
40063       40063  1461639063329550344   
2252         2252  1461451876000489474   
43940       43940  1461577564850319362   
34370       34370  1461716582602657792   
41984       41984  1461611790907809795   
6960         6960  1461393296190840837   
23267       23267  1461168528757829633   
5082         5082  1461415007846117376   
15812       15812  1461293000001208324   
18817       18817  1461253366483402754   
42346       42346  1461606317567754240   
859           859  1461469510586839047   
3009         3009  1461442074184720387   
40431       40431  1461634086292840448   
10855       10855  1461347421116502019   
15874       15874  1461292250927972353   
41506       41506  1461619083162664963   
32104       32104  1461744173653139459   
5530         5530  1461411825971515395   
32963       32963  1461733446355669006   
40612       40612  1461631578107482112   
32155       32155  1461743601382268936   
5403         5403  1461413368959492108   
16547       16547  1461283995266830338   
4585         4585  1461420287510159373   
5484         5484  1461412419377537025   
6012         6012  1461405606540103682   
6999         6999  1461392799459385347   
18871       18871  1461252721122721793   
42229       42229  1461608131411255297   
18905       18905  1461252203105206277   
41844       41844  1461613986038919171   
18896       18896  1461252366175453184   
29595       29595  1461776690099728385   
14561       14561  1461307600562266117   
10645       10645  1461349756005806089   
4527         4527  1461421081105948679   
30531       30531  1461764692653887496   
2434         2434  1461450207657349131   
24919       24919  1461131965848068101   
32805       32805  1461735560435752961   
10702       10702  1461349106786332673   
29556       29556  1461777388694654980   
28012       28012  1461802373387591683   
25807       25807  1461840296921210886   
30302       30302  1461767227603046403   
18546       18546  1461257856771137537   
47386       47386  1461487764411764739   
12317       12317  1461331977123024908   
6799         6799  1461395013494099969   
6687         6687  1461396578380484619   
12348       12348  1461331560037335048   
12311       12311  1461332052465250307   
24178       24178  1461146737670864901   
45346       45346  1461538703545884672   
22596       22596  1461185651815202816   
11775       11775  1461337332011962376   
46614       46614  1461503200280788992   
47400       47400  1461487406486458370   
11205       11205  1461343665352228868   
12339       12339  1461331701699948548   
12900       12900  1461325554641256451   
12343       12343  1461331621559386113   
6708         6708  1461396265426636806   
12332       12332  1461331761800126473   
11232       11232  1461343269153951751   
12320       12320  1461331956629704712   
9246         9246  1461366026193362954   
11731       11731  1461337836246863873   
12347       12347  1461331576944537607   
6759         6759

### <font color='green'>Answer(last part): Tweets shown above have largest level of amplification volume of this dataset. Tweets that have same amplification_volume are same retweeted tweets by different accounts.</font>

## Question 2. Analysis of highly amplified conversations (15 points)

### Please describe  5 highly amplified conversations which are trying to polarize or misinform.

## Tip: given a tweet.id it is possible to have a detailed look at twitter by appending this id to the url: https://twitter.com/anyuser/status/


### For instance given the tweet.id: 1461577564850319362 you can inspect its content by having a look at: https://twitter.com/anyuser/status/1461577564850319362



In [12]:
grouped_tweet_text = spainEnglishAll.groupby(["tweet.text"]).mean("amplification_volume").sort_values("amplification_volume", ascending=False)

In [13]:
grouped_tweet_text.shape

(22785, 6)

In [14]:
22785*2 #means that, there are some tweets got retweet from different accounts in these dataset.

45570

###  <font color='green'> This result means that, there are some tweets got retweet from different accounts in this dataset.</font>

In [15]:
grouped_tweet_text.head(25)

Unnamed: 0  \
tweet.text                                                       
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       23467   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       16238   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...       39357   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       36979   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45333   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45374   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       33960   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45332   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       45336   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       34516   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\...       15299   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       45334   
RT @NCTsmtown_127: Amazing time in SPAIN🇪🇸\nGra...       23461   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       43604   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       45337   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\...       45335   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       40063   
RT @MesutOzil1088: I grew up as a @Fenerbahce f...        2252   
RT @LAPXURADELBUIO: what is happening in spain ...       43940   
RT @AfricanArchives: African moors, who conquer...       34370   
RT @TheeAzanian: Normalize posting these poor p...       24472   
RT @bubbaprog: Here is how the Nicki Minaj stor...       23267   
RT @viquirepublica: 📌This is the real problem i...       19793   
EURO 2008 top scorer: Villa ⚽️\nWorld Cup 2010 ...        2434   
RT @MaziNnamdiKanu: Gerard Pique is a World Cup...       24919   

                                                       tweet.author_id  \
tweet.text                                                               
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1290910527393210368   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...  550915711900380928   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC... 1064006291087970432   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1290852263092957184   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1306812996296015872   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1382234880465989632   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1242033151548897280   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\... 1148751069979847296   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1382234880465989632   
RT @NCTsmtown_127: Amazing time in SPAIN🇪🇸\nGra... 1290910527393210368   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1385362683323533312   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1382234880465989632   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1430829012520214528   
RT @MesutOzil1088: I grew up as a @Fenerbahce f... 1245113611263057920   
RT @LAPXURADELBUIO: what is happening in spain ... 1324927883446345728   
RT @AfricanArchives: African moors, who conquer... 1366298396051582976   
RT @TheeAzanian: Normalize posting these poor p... 1332043128319223808   
RT @bubbaprog: Here is how the Nicki Minaj stor...          1138552825   
RT @viquirepublica: 📌This is the real problem i...  530809508266492672   
EURO 2008 top scorer: Villa ⚽️\nWorld Cup 2010 ...            18091004   
RT @MaziNnamdiKanu: Gerard Pique is a World Cup... 1416473790775795712   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         44845   
RT @NCTsmtown_127:

In [16]:
NCTsmtown_event = grouped_tweet_text[0:17]

In [17]:
NCTsmtown_event

Unnamed: 0  \
tweet.text                                                       
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       23467   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       16238   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...       39357   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       36979   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45333   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45374   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       33960   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45332   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       45336   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       34516   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\...       15299   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       45334   
RT @NCTsmtown_127: Amazing time in SPAIN🇪🇸\nGra...       23461   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       43604   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       45337   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\...       45335   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...       40063   

                                                       tweet.author_id  \
tweet.text                                                               
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1290910527393210368   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...  550915711900380928   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC... 1064006291087970432   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1290852263092957184   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1306812996296015872   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1382234880465989632   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1242033151548897280   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\... 1148751069979847296   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1382234880465989632   
RT @NCTsmtown_127: Amazing time in SPAIN🇪🇸\nGra... 1290910527393210368   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1385362683323533312   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1382234880465989632   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\... 1382234880465989632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain... 1430829012520214528   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         44845   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         43945   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...                         35930   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         34538   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         32878   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         32632   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...                         30490   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         30038   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...                         27322   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         27002   
RT @NCTsmtown_127: #Johnny's Photo Diary:Spain\...                         26867   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...                         26424   
RT @NCTsmtown_127: Amazing time in SPAIN🇪🇸\nGra...                         25251   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spain...                         22858   
RT @NCTsmtown_127: #Johnny 's Photo Diary:Spai

### <font color='green'> The biggest conversation that has large amplification of misinformation is NCT 127 pop group in given tweets or dates. Some accounts share old pictures of the group, in order to show them like they are in Spain. According to https://www.newsnow.co.uk/h/Entertainment+&+Arts/Music/K-Pop/SM+Entertainment/NCT, group was in USA at given dates. These conversations can be part of marketing operation by bots or fake accounts. Aim of operation can be increase the popularity of the group.</font>

In [18]:
spainEnglishAll.iloc[2252]

Unnamed: 0                                                                   2252
tweeet.id                                                     1461451876000489474
tweet.text                      RT @MesutOzil1088: I grew up as a @Fenerbahce ...
tweet.created_at                                        2021-11-18 21:50:54+00:00
tweet.author_id                                               1245113611263057920
tweet.lang                                                                     en
public_metrics.retweet_count                                                17801
public_metrics.reply_count                                                      0
public_metrics.like_count                                                       0
amplification_volume                                                        89005
Name: 2252, dtype: object

In [19]:
grouped_tweet_text[17:18]

Unnamed: 0  \
tweet.text                                                       
RT @MesutOzil1088: I grew up as a @Fenerbahce f...        2252   

                                                       tweet.author_id  \
tweet.text                                                               
RT @MesutOzil1088: I grew up as a @Fenerbahce f... 1245113611263057920   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @MesutOzil1088: I grew up as a @Fenerbahce f...                         17801   

                                                    public_metrics.reply_count  \
tweet.text                                                                       
RT @MesutOzil1088: I grew up as a @Fenerbahce f...                           0   

                                                    public_metrics.like_count  \
tweet.text                                                                      
RT @MesutOzil1088: I grew up as a @Fenerbahce f...                          0   

                                                    amplification_volume  
tweet.text                                                                
RT @MesutOzil1088: I grew up as a @Fenerbahce f...                 89005

###  <font color='green'>Second conversation that had big amplification volume is a tweet that can used by people who want to polarise the country. In these tweet, Mesut Ozil who is a famous football player, claim that Real Madrid is the biggest team of Spain and by saying that he created suitable opportunity to people who can increase the polarisation in Spain.</font>

In [20]:
spainEnglishAll.iloc[43940]

Unnamed: 0                                                                  43940
tweeet.id                                                     1461577564850319362
tweet.text                      RT @LAPXURADELBUIO: what is happening in spain...
tweet.created_at                                        2021-11-19 06:10:20+00:00
tweet.author_id                                               1324927883446345728
tweet.lang                                                                     en
public_metrics.retweet_count                                                16466
public_metrics.reply_count                                                      0
public_metrics.like_count                                                       0
amplification_volume                                                        82330
Name: 43940, dtype: object

In [21]:
grouped_tweet_text[18:19]

Unnamed: 0  \
tweet.text                                                       
RT @LAPXURADELBUIO: what is happening in spain ...       43940   

                                                       tweet.author_id  \
tweet.text                                                               
RT @LAPXURADELBUIO: what is happening in spain ... 1324927883446345728   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @LAPXURADELBUIO: what is happening in spain ...                         16466   

                                                    public_metrics.reply_count  \
tweet.text                                                                       
RT @LAPXURADELBUIO: what is happening in spain ...                           0   

                                                    public_metrics.like_count  \
tweet.text                                                                      
RT @LAPXURADELBUIO: what is happening in spain ...                          0   

                                                    amplification_volume  
tweet.text                                                                
RT @LAPXURADELBUIO: what is happening in spain ...                 82330

### <font color='green'>Third conversation that had big amplification volume in these dataset is a retweet of a tweet that published in 5 Jul 2021. This tweet emphasise that Spanish policemen are targeting Lgbtq2+ community in Spain. This post is clearly misinformation about Spain.</font>

In [22]:
spainEnglishAll.iloc[34370]

Unnamed: 0                                                                  34370
tweeet.id                                                     1461716582602657792
tweet.text                      RT @AfricanArchives: African moors, who conque...
tweet.created_at                                        2021-11-19 15:22:45+00:00
tweet.author_id                                               1366298396051582976
tweet.lang                                                                     en
public_metrics.retweet_count                                                16115
public_metrics.reply_count                                                      0
public_metrics.like_count                                                       0
amplification_volume                                                        80575
Name: 34370, dtype: object

In [23]:
grouped_tweet_text[19:20]

Unnamed: 0  \
tweet.text                                                       
RT @AfricanArchives: African moors, who conquer...       34370   

                                                       tweet.author_id  \
tweet.text                                                               
RT @AfricanArchives: African moors, who conquer... 1366298396051582976   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @AfricanArchives: African moors, who conquer...                         16115   

                                                    public_metrics.reply_count  \
tweet.text                                                                       
RT @AfricanArchives: African moors, who conquer...                           0   

                                                    public_metrics.like_count  \
tweet.text                                                                      
RT @AfricanArchives: African moors, who conquer...                          0   

                                                    amplification_volume  
tweet.text                                                                
RT @AfricanArchives: African moors, who conquer...                 80575

### <font color='green'>Fourth conversation that has big amplification volume is about history. In this conversation, some account claim that black African people introduce education to the European people. This claim is misinformation. Because people who invaded Spain was leading by Arabic dynasty. In addition to that, even Arabic state had further education quality against Europeans, their effects cannot be only reason of increasing education level in West.</font>

In [24]:
spainEnglishAll.iloc[41984]

Unnamed: 0                                                                  41984
tweeet.id                                                     1461611790907809795
tweet.text                      RT @TheeAzanian: Normalize posting these poor ...
tweet.created_at                                        2021-11-19 08:26:21+00:00
tweet.author_id                                               1412044637288779776
tweet.lang                                                                     en
public_metrics.retweet_count                                                13620
public_metrics.reply_count                                                      0
public_metrics.like_count                                                       0
amplification_volume                                                        68100
Name: 41984, dtype: object

In [25]:
grouped_tweet_text[20:21]

Unnamed: 0  \
tweet.text                                                       
RT @TheeAzanian: Normalize posting these poor p...       24472   

                                                       tweet.author_id  \
tweet.text                                                               
RT @TheeAzanian: Normalize posting these poor p... 1332043128319223808   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @TheeAzanian: Normalize posting these poor p...                         13620   

                                                    public_metrics.reply_count  \
tweet.text                                                                       
RT @TheeAzanian: Normalize posting these poor p...                           0   

                                                    public_metrics.like_count  \
tweet.text                                                                      
RT @TheeAzanian: Normalize posting these poor p...                          0   

                                                    amplification_volume  
tweet.text                                                                
RT @TheeAzanian: Normalize posting these poor p...                 68100

### <font color='green'>Fifth biggest conversation is a retweet of a tweet like previous conversations. This tweet has misinformation. Because people in the pictures claimed that they are from Spain and they are living in very poor conditions. However, people in the reply section say that they are refugees or this pictures have taken in some Asian country. There is no clear evidence that where is the pictures have taken and who the settlers are in the pictures. </font>

## Question 3. Compute the z-score of the column "public_metrics.retweet_count" (15 points)
## tip: refer to the definition of Z-score at" https://www.statisticshowto.com/probability-and-statistics/z-score/

In [26]:
spainEnglishAll

Unnamed: 0            tweeet.id  \
0               0  1461484086892146692   
1               1  1461484079245840390   
2               2  1461484043187462149   
3               3  1461483982105763842   
4               4  1461483978658127872   
...           ...                  ...   
47545       47545  1461484434209783814   
47546       47546  1461484401712414722   
47547       47547  1461484391255977994   
47548       47548  1461484374558490625   
47549       47549  1461484366325035009   

                                              tweet.text  \
0      RT @ESPNFC: EURO 2008 top scorer: Villa ⚽️\nWo...   
1      RT @MrGeepGeep: SOE is over and now is the tim...   
2      RT @AnittaCrave: "Envolver" — by Anitta will d...   
3      Men jewelry  - Madrid, Spain \n@GuilleRBRB htt...   
4      RT @Foreign_Cat: #Turkey &amp; #Spain, partner...   
...                                                  ...   
47545  @only1womble @SimonJonesNews @pritipatel Someo...   
47546  RT @ancientorigins: A team of #scientists has ...   
47547  On #Friday Amazing time in #SabatiniGardens #M...   
47548  RT @THEMAXHILTON: The wait is over !  you have...   
47549  #Barcelona Nov 19 01:00 Temperature 10C Wind N...   

                tweet.created_at     tweet.author_id tweet.lang  \
0      2021-11-18 23:58:54+00:00          3239744421         en   
1      2021-11-18 23:58:52+00:00          2312771653         en   
2      2021-11-18 23:58:43+00:00 1296857652677550080         en   
3      2021-11-18 23:58:29+00:00 1071703322166480768         en   
4      2021-11-18 23:58:28+00:00 1142741725225652224         en   
...                          ...                 ...        ...   
47545  2021-11-19 00:00:16+00:00          1733373504         en   
47546  2021-11-19 00:00:09+00:00           100887413         en   
47547  2021-11-19 00:00:06+00:00  995441471938580352         en   
47548  2021-11-19 00:00:02+00:00  832015137263452160         en   
47549  2021-11-19 00:00:00+00:00           258025521         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
0                              1568                           0   
1                                38                           0   
2                                90                           0   
3                                 0                           0   
4                                34                           0   
...                             ...                         ...   
47545                             0                           0   
47546                            18                           0   
47547                             0                           0   
47548                           121                           0   
47549                             0                           0   

       public_metrics.like_count  amplification_volume  
0                              0                  7840  
1                              0                   190  
2                              0                   450  
3                              5                    10  
4                              0                   170  
...                          ...                   ...  
47545                          0                     0  
47546                          0                    90  
47547                          1                     2  
47548                          0                   605  
47549                          0                     0  

[47550 rows x 10 columns]

In [27]:
np_retweet_counts = np.array(spainEnglishAll["public_metrics.retweet_count"])

In [28]:
np.nanmean(np_retweet_counts)

233.72148011022992

In [29]:
np.nanstd(np_retweet_counts)

959.4851708357776

In [30]:
z_score = (np_retweet_counts-np.nanmean(np_retweet_counts)) / np.nanstd(np_retweet_counts)

In [31]:
z_score

array([ 1.39061922, -0.20398594, -0.1497902 , ..., -0.24359051,
       -0.11748121, -0.24359051])

In [32]:
z_score.shape

(47550,)

In [33]:
z_score[0]

1.3906192200214222

In [34]:
spainEnglishAll["z_score_retweet"] = (spainEnglishAll["public_metrics.retweet_count"]-spainEnglishAll["public_metrics.retweet_count"].mean()) / spainEnglishAll["public_metrics.retweet_count"].std()

In [35]:
spainEnglishAll

Unnamed: 0            tweeet.id  \
0               0  1461484086892146692   
1               1  1461484079245840390   
2               2  1461484043187462149   
3               3  1461483982105763842   
4               4  1461483978658127872   
...           ...                  ...   
47545       47545  1461484434209783814   
47546       47546  1461484401712414722   
47547       47547  1461484391255977994   
47548       47548  1461484374558490625   
47549       47549  1461484366325035009   

                                              tweet.text  \
0      RT @ESPNFC: EURO 2008 top scorer: Villa ⚽️\nWo...   
1      RT @MrGeepGeep: SOE is over and now is the tim...   
2      RT @AnittaCrave: "Envolver" — by Anitta will d...   
3      Men jewelry  - Madrid, Spain \n@GuilleRBRB htt...   
4      RT @Foreign_Cat: #Turkey &amp; #Spain, partner...   
...                                                  ...   
47545  @only1womble @SimonJonesNews @pritipatel Someo...   
47546  RT @ancientorigins: A team of #scientists has ...   
47547  On #Friday Amazing time in #SabatiniGardens #M...   
47548  RT @THEMAXHILTON: The wait is over !  you have...   
47549  #Barcelona Nov 19 01:00 Temperature 10C Wind N...   

                tweet.created_at     tweet.author_id tweet.lang  \
0      2021-11-18 23:58:54+00:00          3239744421         en   
1      2021-11-18 23:58:52+00:00          2312771653         en   
2      2021-11-18 23:58:43+00:00 1296857652677550080         en   
3      2021-11-18 23:58:29+00:00 1071703322166480768         en   
4      2021-11-18 23:58:28+00:00 1142741725225652224         en   
...                          ...                 ...        ...   
47545  2021-11-19 00:00:16+00:00          1733373504         en   
47546  2021-11-19 00:00:09+00:00           100887413         en   
47547  2021-11-19 00:00:06+00:00  995441471938580352         en   
47548  2021-11-19 00:00:02+00:00  832015137263452160         en   
47549  2021-11-19 00:00:00+00:00           258025521         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
0                              1568                           0   
1                                38                           0   
2                                90                           0   
3                                 0                           0   
4                                34                           0   
...                             ...                         ...   
47545                             0                           0   
47546                            18                           0   
47547                             0                           0   
47548                           121                           0   
47549                             0                           0   

       public_metrics.like_count  amplification_volume  z_score_retweet  
0                              0                  7840                1  
1                              0                   190               -0  
2                              0                   450               -0  
3                              5                    10               -0  
4                              0                   170               -0  
...                          ...                   ...              ...  
47545                          0                     0               -0  
47546                          0                    90               -0  
47547                          1                     2               -0  
47548                          0                   605               -0  
47549                          0                     0               -0  

[47550 rows x 11 columns]

## Question 4. Filter conversations with a z-score larger than 10. (15 points)

In [36]:
filter_z_score = []

In [37]:
for i in z_score:
    if i>10:
        filter_z_score.append(i)
print(filter_z_score)

[10.083822881245718, 18.309067251750708, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 13.951522052424636, 10.083822881245718, 45.557013123834146, 27.757884467031786, 10.083822881245718, 10.083822881245718, 27.757884467031786, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 12.927014295682822, 27.757884467031786, 26.073647910679014, 31.532825560540285, 46.49501615645636, 35.75175475615663, 45.557013123834146, 27.897542696333314, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 10.083822881245718, 16.55187490397177, 37.20357500553745, 22.11944401538049, 10.083822881245718, 10.083822881245718, 10.083822881245718, 23.579602069495735, 10.083822881245718, 35.75279698174843, 13.951522052424636, 10.083822881245718, 10.083822881245718, 16.91

In [38]:
len(filter_z_score)

62

In [39]:
answer_four = spainEnglishAll[spainEnglishAll["z_score_retweet"]>10.00]

In [40]:
answer_four.shape

(62, 11)

In [41]:
answer_four

Unnamed: 0            tweeet.id  \
859           859  1461469510586839047   
2252         2252  1461451876000489474   
3009         3009  1461442074184720387   
4527         4527  1461421081105948679   
4585         4585  1461420287510159373   
5082         5082  1461415007846117376   
5403         5403  1461413368959492108   
5484         5484  1461412419377537025   
5530         5530  1461411825971515395   
6012         6012  1461405606540103682   
6960         6960  1461393296190840837   
6999         6999  1461392799459385347   
8820         8820  1461371211343020033   
9806         9806  1461359523298377745   
10645       10645  1461349756005806089   
10855       10855  1461347421116502019   
12779       12779  1461327131057680386   
14561       14561  1461307600562266117   
15812       15812  1461293000001208324   
15874       15874  1461292250927972353   
16547       16547  1461283995266830338   
18817       18817  1461253366483402754   
18871       18871  1461252721122721793   
18896       18896  1461252366175453184   
18905       18905  1461252203105206277   
23267       23267  1461168528757829633   
23311       23311  1461167495809617921   
23461       23461  1461162923057827840   
23465       23465  1461162702307426312   
23467       23467  1461162679809155076   
23655       23655  1461157789397438464   
23656       23656  1461157780664901633   
23658       23658  1461157760607739909   
29595       29595  1461776690099728385   
30531       30531  1461764692653887496   
32104       32104  1461744173653139459   
32155       32155  1461743601382268936   
32963       32963  1461733446355669006   
34370       34370  1461716582602657792   
39357       39357  1461649200882405377   
40063       40063  1461639063329550344   
40431       40431  1461634086292840448   
40612       40612  1461631578107482112   
41506       41506  1461619083162664963   
41830       41830  1461614169296306178   
41844       41844  1461613986038919171   
41908       41908  1461613145219239944   
41984       41984  1461611790907809795   
42229       42229  1461608131411255297   
42346       42346  1461606317567754240   
43940       43940  1461577564850319362   
44456       44456  1461565440879206401   
45332       45332  1461539163207995397   
45333       45333  1461539133520695297   
45334       45334  1461539103741149186   
45335       45335  1461539081725251585   
45336       45336  1461539044974731265   
45337       45337  1461539030298886147   
45373       45373  1461537918900600843   
45374       45374  1461537902106537991   
45375       45375  1461537878991781892   
45378       45378  1461537845516984321   

                                              tweet.text  \
859    RT @viquirepublica: 📌This is the real problem ...   
2252   RT @MesutOzil1088: I grew up as a @Fenerbahce ...   
3009   RT @viquirepublica: 📌This is the real problem ...   
4527   RT @viquirepublica: 📌This is the real problem ...   
4585   RT @viquirepublica: 📌This is the real problem ...   
5082   RT @viquirepublica: 📌This is the real problem ...   
5403   RT @viquirepublica: 📌This is the real problem ...   
5484   RT @viquirepublica: 📌This is the real problem ...   
5530   RT @viquirepublica: 📌This is the real problem ...   
6012   RT @viquirepublica: 📌This is the real problem ...   
6960   RT @TheeAzanian: Normalize posting these poor ...   
6999   RT @viquirepublica: 📌This is the real problem ...   
8820   RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain...   
9806   RT @NCTsmtown_127: #Johnny's Photo Diary:Spain...   
10645  RT @viquirepublica: 📌This is the real problem ...   
10855  RT @viquirepublica: 📌This is the real problem ...   
12779  RT @NCTsmtown_127: #Johnny's Photo Diary:Spain...   
14561  RT @viquirepublica: 📌This is the real problem ...   
15812  RT @viquirepublica: 📌This is the real problem ...   
15874  RT @viquirepublica: 📌This is the real problem ...   
16547  RT @viquirepublica: 📌This is the real problem ...   
18817  RT @viquirepublica: 📌This is the 

## Question 5. Find the top 5 amplifiers (15 points)

### <font color='green'>We consider amplifiers those twitter users who are involved in highly amplified conversations.</font>

In [42]:
spainEnglishAll.groupby(["tweet.author_id"]).mean("amplification_volume").sort_values("amplification_volume", ascending=False).head(5)

Unnamed: 0  public_metrics.retweet_count  \
tweet.author_id                                                 
1168957082                 8820                         43945   
1064006291087970432       39357                         35930   
1101831422631804800       23656                         35161   
1290910527393210368       23426                         31863   
1322715465198821376       44456                         30490   

                     public_metrics.reply_count  public_metrics.like_count  \
tweet.author_id                                                              
1168957082                                    0                          0   
1064006291087970432                           0                          0   
1101831422631804800                           0                          0   
1290910527393210368                           0                          0   
1322715465198821376                           0                          0   

                     amplification_volume  z_score_retweet  
tweet.author_id                                             
1168957082                         219725               46  
1064006291087970432                179650               37  
1101831422631804800                175805               36  
1290910527393210368                159315               33  
1322715465198821376                152450               32

### <font color='green'>Answer: For this question, we need to look average number of amplification of every tweets for per tweet.author_id. If the number is high, these accounts only retweet, like or reply in high amplified conversations in this given dataset.</font>

In [43]:
grouped_tweet_text.head(5)

Unnamed: 0  \
tweet.text                                                       
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       23467   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       16238   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...       39357   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       36979   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...       45333   

                                                       tweet.author_id  \
tweet.text                                                               
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1290910527393210368   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...  550915711900380928   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC... 1064006291087970432   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1290852263092957184   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\... 1382234880465989632   

                                                    public_metrics.retweet_count  \
tweet.text                                                                         
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         44845   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         43945   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...                         35930   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         34538   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                         32878   

                                                    public_metrics.reply_count  \
tweet.text                                                                       
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                           0   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                           0   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...                           0   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                           0   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                           0   

                                                    public_metrics.like_count  \
tweet.text                                                                      
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                          0   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                          0   
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...                          0   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                          0   
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                          0   

                                                    amplification_volume  
tweet.text                                                                
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                224225  
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                219725  
RT @NCTsmtown_127: Spain here we goooooo\n\n#NC...                179650  
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                172688  
RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain\...                164390

In [44]:
spainEnglishAll[spainEnglishAll["tweet.author_id"]==1168957082]

Unnamed: 0            tweeet.id  \
8820        8820  1461371211343020033   

                                             tweet.text  \
8820  RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain...   

               tweet.created_at  tweet.author_id tweet.lang  \
8820  2021-11-18 16:30:22+00:00       1168957082         en   

      public_metrics.retweet_count  public_metrics.reply_count  \
8820                         43945                           0   

      public_metrics.like_count  amplification_volume  z_score_retweet  
8820                          0                219725               46

In [45]:
spainEnglishAll[spainEnglishAll["tweet.author_id"]==1064006291087970432]

Unnamed: 0            tweeet.id  \
39357       39357  1461649200882405377   

                                              tweet.text  \
39357  RT @NCTsmtown_127: Spain here we goooooo\n\n#N...   

                tweet.created_at     tweet.author_id tweet.lang  \
39357  2021-11-19 10:55:00+00:00 1064006291087970432         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
39357                         35930                           0   

       public_metrics.like_count  amplification_volume  z_score_retweet  
39357                          0                179650               37

In [46]:
spainEnglishAll[spainEnglishAll["tweet.author_id"]==1101831422631804800]

Unnamed: 0            tweeet.id  \
23655       23655  1461157789397438464   
23656       23656  1461157780664901633   
23658       23658  1461157760607739909   

                                              tweet.text  \
23655  RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain...   
23656  RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain...   
23658  RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain...   

                tweet.created_at     tweet.author_id tweet.lang  \
23655  2021-11-18 02:22:18+00:00 1101831422631804800         en   
23656  2021-11-18 02:22:16+00:00 1101831422631804800         en   
23658  2021-11-18 02:22:11+00:00 1101831422631804800         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
23655                         34537                           0   
23656                         43945                           0   
23658                         27001                           0   

       public_metrics.like_count  amplification_volume  z_score_retweet  
23655                          0                172685               36  
23656                          0                219725               46  
23658                          0                135005               28

In [47]:
spainEnglishAll[spainEnglishAll["tweet.author_id"]==1290910527393210368]

Unnamed: 0            tweeet.id  \
23311       23311  1461167495809617921   
23461       23461  1461162923057827840   
23465       23465  1461162702307426312   
23467       23467  1461162679809155076   

                                              tweet.text  \
23311  RT @NCTsmtown_127: #Johnny's Photo Diary:Spain...   
23461  RT @NCTsmtown_127: Amazing time in SPAIN🇪🇸\nGr...   
23465  RT @NCTsmtown_127: #Johnny 's Photo Diary:Spai...   
23467  RT @NCTsmtown_127: #Johnny’s Photo Diary:Spain...   

                tweet.created_at     tweet.author_id tweet.lang  \
23311  2021-11-18 03:00:52+00:00 1290910527393210368         en   
23461  2021-11-18 02:42:42+00:00 1290910527393210368         en   
23465  2021-11-18 02:41:49+00:00 1290910527393210368         en   
23467  2021-11-18 02:41:44+00:00 1290910527393210368         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
23311                         26867                           0   
23461                         25251                           0   
23465                         30489                           0   
23467                         44845                           0   

       public_metrics.like_count  amplification_volume  z_score_retweet  
23311                          0                134335               28  
23461                          0                126255               26  
23465                          0                152445               32  
23467                          0                224225               46

In [48]:
spainEnglishAll[spainEnglishAll["tweet.author_id"]==1322715465198821376]

Unnamed: 0            tweeet.id  \
44456       44456  1461565440879206401   

                                              tweet.text  \
44456  RT @NCTsmtown_127: #Johnny 's Photo Diary:Spai...   

                tweet.created_at     tweet.author_id tweet.lang  \
44456  2021-11-19 05:22:10+00:00 1322715465198821376         en   

       public_metrics.retweet_count  public_metrics.reply_count  \
44456                         30490                           0   

       public_metrics.like_count  amplification_volume  z_score_retweet  
44456                          0                152450               32

###  <font color='green'>As we can see, tweet.author_id numbers 1290910527393210368 and 1101831422631804800 have joined highest amplified conversations, but they also have joined less amplified conversations too. Thus, their average amplification_volume got decrease.</font>

## Question 6. Extract, from the text colum, information on who created the conversation in the first place (15 points)
### Tip: retweeted conversations start with the pattern RT@
### For illustration purposes, the text "RT @20splentyforus: France, like UK should set a national 30km/h or 20mph urban default limit." was originally created by @20splentyforus
### Tip2. consider applying a regex pattern to the DataFrame using apply method

In [49]:
spainEnglishAll["tweet.text"].dtype

dtype('O')

In [50]:
spainEnglishAll["public_metrics.retweet_count"].dtype

dtype('float64')

In [51]:
tweet_text_str = spainEnglishAll["tweet.text"].astype(str)

In [52]:
original_accountt = tweet_text_str.str.extract(r"(^RT.@[a-zA-Z|\d]*.[a-zA-Z|\d]*)", expand=False)

In [53]:
original_accountt

0                RT @ESPNFC:
1            RT @MrGeepGeep:
2           RT @AnittaCrave:
3                        NaN
4            RT @Foreign_Cat
                ...         
47545                    NaN
47546    RT @ancientorigins:
47547                    NaN
47548      RT @THEMAXHILTON:
47549                    NaN
Name: tweet.text, Length: 47550, dtype: object

In [54]:
spainEnglishAll["original_account"] = original_accountt

In [55]:
spainEnglishAll.loc["34123":"34234"]

Unnamed: 0            tweeet.id  \
34123       34123  1461719597602672643   
34124       34124  1461719594046038017   
34125       34125  1461719585166667776   
34126       34126  1461719583862177792   
34127       34127  1461719564551528451   
34128       34128  1461719560449626125   
34129       34129  1461719540786733064   
34130       34130  1461719508901666822   
34131       34131  1461719502257889290   
34132       34132  1461719483471573003   
34133       34133  1461719464408494086   
34134       34134  1461719441054466048   
34135       34135  1461719378614009856   
34136       34136  1461719355209576452   
34137       34137  1461719347957743616   
34138       34138  1461719347697643528   
34139       34139  1461719313723924481   
34140       34140  1461719300008517635   
34141       34141  1461719292823625728   
34142       34142  1461719290827231241   
34143       34143  1461719273966084106   
34144       34144  1461719269209690116   
34145       34145  1461719262939258881   
34146       34146  1461719253997002760   
34147       34147  1461719252696813574   
34148       34148  1461719234673790976   
34149       34149  1461719221252022286   
34150       34150  1461719216164417542   
34151       34151  1461719206324420611   
34152       34152  1461719194802827275   
34153       34153  1461719192407875589   
34154       34154  1461719189450747905   
34155       34155  1461719184946221061   
34156       34156  1461719134035746825   
34157       34157  1461719129547808773   
34158       34158  1461719105728180230   
34159       34159  1461719037470150657   
34160       34160  1461719036648210441   
34161       34161  1461719035792568335   
34162       34162  1461719028129570818   
34163       34163  1461719025411670020   
34164       34164  1461719020856619014   
34165       34165  1461719015571742741   
34166       34166  1461718985238622221   
34167       34167  1461718984919769101   
34168       34168  1461718975805493248   
34169       34169  1461718962161418244   
34170       34170  1461718959204536324   
34171       34171  1461718950572699648   
34172       34172  1461718941936586765   
34173       34173  1461718927277502465   
34174       34174  1461718925679415304   
34175       34175  1461718924458872833   
34176       34176  1461718903604797445   
34177       34177  1461718896734580737   
34178       34178  1461718893995638786   
34179       34179  1461718883438583811   
34180       34180  1461718883069571077   
34181       34181  1461718869891067905   
34182       34182  1461718847623512075   
34183       34183  1461718846922911744   
34184       34184  1461718791428182021   
34185       34185  1461718778455236615   
34186       34186  1461718774047023114   
34187       34187  1461718766568443906   
34188       34188  1461718739783720979   
34189       34189  1461718739116769282   
34190       34190  1461718729423790091   
34191       34191  1461718714932416513   
34192       34192  1461718702995349507   
34193       34193  1461718699606487046   
34194       34194  1461718698784395269   
34195       34195  1461718697408704525   
34196       34196  1461718689657536520   
34197       34197  1461718681709424642   
34198       34198  1461718651485110276   
34199       34199  1461718633839747072   
34200       34200  1461718629582581760   
34201       34201  1461718621256917001   
34202       34202  1461718615946891264   
34203       34203  1461718591892561932   
34204       34204  1461718577158017032   
34205       34205  1461718575690010631   
34206       34206  1461718572779122698   
34207       34207  1461718564021305351   
34208       34208  1461718553950887944   
34209       34209  1461718552923279365   
34210       34210  1461718548066095114   
34211       34211  1461718529229664257   
34212       34212  1461718505716424707   
34213       34213  1461718495813644294   
34214       34214  1461718482345730057   
34215       34215  1461718481246867460   
34216       34216  1461718477492965379   
34217       34217

### <font color='green'>Answer: original_account column has the account names who have started the conversations.</font>

## Question 7. Please briefly reflect on how would you try to mitigate misinformation on social media (2 paragraphs maximum). (10 points)

####  We are living in communication era, so number of conversations have dramatically increased. This improvement provides great individual power that ability to increase people’ voice, also birth to potential risks such as suitable environment for spreading misinformation. I believe that there are two aspects of solution in this problem, which are external aspect and internal aspect. The solution can be only reached if these two sides work together. External aspect includes regulators such as states and international institutes that has authority to declare rule for mitigating misinformation. External factors can create standard misinformation organizations in a global and regional scales, so they can do necessary research and determine scientific steps to prevent spreading of misinformation by auditing social media companies.  

#### I believe one solution cannot be enough for this problem, so private companies have to try to prevent this problem too. My second potential solution that can help other solutions, is that companies can grade the posts rather than profiles. To illustrate, if someone has a claim about the history, he or she needs to show source to make his or her post validation sign. It does not matter where the person gets source in order to get validation sign. Person just need to show the source and then directly gets validation sign. Even the source is wrong at least people can check the trust level of the source easier than post that has 240 characters from unknown person. 